### ISSUE TITLE
Spatie\Ray\Client doesn't use Craft settings correctly

### ISSUE THREAD

**Issue by juban** [spatie/craft-ray#4]
**Describe the bug**
Ray does not seem to work properly when `host` and `port` settings are defined as non default values in Craft settings.
For instance, if I set the `host` parameter as `host.docker.internal` value, either from the settings panel or in a configuration file, the client still make requests by calling `localhost` because `Spatie\Ray\Client` still get default settings values instead of the Craft ones. Hard coding `$this->host = 'host.docker.internal';` at the beginning of the `getCurlHandleForUrl` method make `ray` behave as expected.

**Versions**
Ray version: 1.12.0
- spatie/craft-ray package version: 1.1.3
- Craft CMS version: 3.6.5.1
- PHP version: 7.4 (using ddev 1.16.5 web image)

**To Reproduce**
1. Change `host` or `port` plugin settings to something else than localhost and 23517
2. Make a call to ray, either from a module, plugin or template
3. Ray does not get the related request and show nothing

**Expected behavior**
Craft plugin should be able to send Ray requests with custom `host` and `port` settings values.

**Desktop:**
 - OS: macOS Big Sur
 - Version 11.2.1

**Comment by davist11** [comment]
Just ran into this problem too and could not figure out what was going on! I was using the `ray.php` method for my settings:

```php
<?php

use craft\helpers\App;

return [
    'host' => 'host.docker.internal',
    'port' => 23517,
    'remote_path' => '/var/www/html',
    'local_path' => App::env('RAY_LOCAL_PATH'),
];
```

And here's what I get when I `var_dump($this->settings);` inside `Spatie\Ray\Ray`

![image](https://user-images.githubusercontent.com/199985/108773584-90021b00-752c-11eb-9108-57c1aaca5495.png)

**Comment by davist11** [comment]
Is there any timeline for a fix here? This makes Ray completely unusable when we are using Docker.

**Comment by riasvdv** [comment]
Should be fixed in 1.1.4!

**Comment by juban** [comment]
@riasvdv Thanks, I will try this ASAP.

**Comment by davist11** [comment]
Hmm I'm still seeing an issue. For one thing, I realized I was using a `config/ray.php` which definitely previously worked. But then I updated to `config/craft-ray.php` and confirmed that the `CraftRay.php` file is correctly grabbing the settings

![image](https://user-images.githubusercontent.com/199985/110142441-2a235800-7da4-11eb-94c2-8625c556627b.png)

Then I threw a `var_dump($this->settings)` into `Spatie\Ray\Ray` and the settings are back to the defaults

![image](https://user-images.githubusercontent.com/199985/110142608-58a13300-7da4-11eb-826e-513d6e9fe13a.png)

**Comment by Diewy** [comment]
Having the same issue when using Ray to debug a Plugin. Ray works fine in Twig functions, but I think the Ray Plugin is loaded _after_ the Plugin I'm debugging and loading the default settings.

Dirty quick fix: Changing the id of the Ray Plugin in the `plugins` table to a lower number.
Or installing Ray as the first Plugin.

Or to reproduce the issue, you can do the opposite.

Note: I'm using [Nitro 2](https://craftcms.com/docs/nitro/2.x/), so a Docker setup.

**Comment by brandonkelly** [comment]
@Diewy You could solve this on your end by putting this in your plugin’s `init()` method:

```php
use craft\web\Application;
use yii\base\Event;

Event::on(Application::class, Application::EVENT_INIT, function() {
    // your initialization code goes here instead...
});
```

Move your current initialization code inside the function, and then it will be sure to be executed _after_ all plugins, including Ray, have been instantiated.

**Comment by riasvdv** [comment]
Closing for now as there's no way to define plugin load order. If it lands in a future version of Craft, we'll take a look at it again.

**Comment by davist11** [comment]
@riasvdv hmm, I'm still experiencing the problem I highlighted above within a module (this used to work). Even if I tried latching onto the event:

```php
Event::on(Application::class, Application::EVENT_INIT, function () {
    ray('hello world');
});
```

**Comment by riasvdv** [comment]
I'll take another look at it this week, if you find a solution in the meantime feel free to PR it

**Comment by davist11** [comment]
Haven't had any chance to look into a solution for this, but have you? Bummer because this was working and we bought a Ray license, but now it's pretty much unusable since we use Docker.

**Comment by riasvdv** [comment]
The plugin is loaded after any custom modules, so the settings won't be applied in that case.

To fix this, you can add a `ray.php` file in the root of your project https://spatie.be/docs/ray/v1/configuration/framework-agnostic-php which will allow you to set the host & port settings on the lowest instance of Ray (the framework agnostic PHP version), which will bubble up to the Craft Ray instance

**Comment by davist11** [comment]
I was using the `ray.php` file before and that stopped working at some point. So that's when I switched to the `craft-ray.php` file and still no dice.

So I switched back to the `config/ray.php` file, and it's still not picking up any of my settings. I dug in a bit and noticed that in `Spatie\Ray\Settings\SettingsFactory` if I output the `$configDirectory` it's `/var/www/html/vendor/craftcms/cms/src/config` which is certainly not right.

Where is it getting that value from?
